solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -16,29 +18,55 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
     }
 
     @Override
-    public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
-    }
+    public void initialize() { }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не может быть null");
+        Objects.requireNonNull(gameState, "Состояние не может быть null");
+
+        try (OutputStream os = new FileOutputStream(filesDirectory + File.separator + filename);
+             ObjectOutputStream oos = new ObjectOutputStream(os)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1613803855>*